stdlib/Distributed/src/managers.jl

```diff
@@ -413,13 +413,20 @@ end
 """
     addprocs(; kwargs...) -> List of process identifiers
 
-Equivalent to `addprocs(Sys.CPU_THREADS; kwargs...)`
+Equivalent to `addprocs(Sys.CPU_THREADS; kwargs...)` if processes have not been launched,
+otherwise it will add only the remaining processes to reach Sys.CPU_THREADS.
 
 Note that workers do not run a `.julia/config/startup.jl` startup script, nor do they synchronize
 their global state (such as global variables, new method definitions, and loaded modules) with any
 of the other running processes.
 """
-addprocs(; kwargs...) = addprocs(Sys.CPU_THREADS; kwargs...)
+function addprocs(; kwargs...)
+    if nprocs() <= Sys.CPU_THREADS 
+        addprocs(Sys.CPU_THREADS-nprocs()+1; kwargs...)
```

Comment:
I guess we could simplify this to:
```jl
addprocs(; kwargs...) = addprocs(max(0, Sys.CPU_THREADS-nprocs()+1); kwargs...)
```
but `addprocs(0)` takes a surprisingly long time on my machine (1 second).

Reply:
`addprocs(0)`, to my knowledge, results in the exact same result as `addprocs(n)` where `n < 0`; If I'm understanding the dispatching correctly, we eventually end up at `for i in 1:manager.np`, and if `1:manager.np < 1` the loop just doesn't iterate through anything.

So if we're going to call `addprocs(0)` at all (and let code execute until this loop), it wouldn't matter if we'd called `addprocs(-1)` or `addprocs(-15)` instead, meaning that the `max` is redundant in this simplification, i.e. we could get the same code execution as your suggestion from this further-simplified chunk:
```
addprocs(; kwargs...) = addprocs(Sys.CPU_THREADS-nprocs()+1; kwargs...)
```

But as you mentioned, while we get the same end result as the original code, both of these "simplifications" result in unnecessary code execution.

This does make me wonder if `addprocs(n)` should instead simply check if `n > 0` and not execute otherwise, since that check should be relatively quick compared to the code that currently always gets executed.